core/src/mill/modules/Jvm.scala

```diff
@@ -61,17 +61,19 @@ object Jvm {
   }
 
 
-
+  /**
+    * run `body` with a isolated classloader
+    */
   def inprocess[T](classPath: Agg[Path],
-                   classLoaderOverrideSbtTesting: Boolean,
+                   classLoaderOverrideSbtTesting: Option[ClassLoader],
                    body: ClassLoader => T): T = {
-    val cl = if (classLoaderOverrideSbtTesting) {
-      val outerClassLoader = getClass.getClassLoader
+    val cl = if (classLoaderOverrideSbtTesting.isDefined) {
+      val outerClassLoader = classLoaderOverrideSbtTesting.get
       new URLClassLoader(
         classPath.map(_.toIO.toURI.toURL).toArray,
         ClassLoader.getSystemClassLoader().getParent()){
         override def findClass(name: String) = {
-          if (name.startsWith("sbt.testing.")){
+          if (name.startsWith("sbt.testing.") || name.startsWith("org.apache.logging.log4j.")){
```

Comment:
Where did this come from btw?

Reply:
Yeah I'd like to understand what we're injecting into the test classpath and why

Reply:
I'm happy to close the PR, but you do not need to wait for ScalaWorker to finalize before fixing the crash log.

The whole work of fixing the issue is understanding what the cause is, in this case what should be in the test classpath, and why your fix is correct. Randomly permuting code for no reason will never make it past code review :)